apis/ecr/v1alpha1/referencers.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+
+	"github.com/crossplane/provider-aws/apis/identity/v1alpha1"
+	identityv1alpha1 "github.com/crossplane/provider-aws/apis/identity/v1alpha1"
+	identityv1beta1 "github.com/crossplane/provider-aws/apis/identity/v1beta1"
+)
+
+// ResolveReferences of this RepositoryPolicy
+func (mg *RepositoryPolicy) ResolveReferences(ctx context.Context, c client.Reader) error {
+	r := reference.NewAPIResolver(c, mg)
+
+	// Resolve spec.forProvider.repositoryName
+	rsp, err := r.Resolve(ctx, reference.ResolutionRequest{
+		CurrentValue: reference.FromPtrValue(mg.Spec.ForProvider.RepositoryName),
+		Reference:    mg.Spec.ForProvider.RepositoryNameRef,
+		Selector:     mg.Spec.ForProvider.RepositoryNameSelector,
+		To:           reference.To{Managed: &Repository{}, List: &RepositoryList{}},
+		Extract:      reference.ExternalName(),
+	})
+	if err != nil {
+		return errors.Wrap(err, "spec.forProvider.RepositoryName")
+	}
+	mg.Spec.ForProvider.RepositoryName = reference.ToPtrValue(rsp.ResolvedValue)
+	mg.Spec.ForProvider.RepositoryNameRef = rsp.ResolvedReference
+
+	// Resolve spec.forProvider.userName
+	if mg.Spec.ForProvider.Policy != nil && mg.Spec.ForProvider.Policy.Statements != nil {
```

Comment:
```suggestion
	if mg.Spec.ForProvider.Policy != nil {
```

`for` would be no-op if array is `nil` or has no elements.